src/geofabrics/dem.py

```diff
@@ -1670,62 +1754,158 @@ def _create_data_set(
             # ensure the expected CF conventions are followed
             self._write_netcdf_conventions_in_place(dem, self.catchment_geometry.crs)
             dems.append(dem)
+
         if len(dems) == 1:
             dem = dems[0]
         else:
             dem = rioxarray.merge.merge_datasets(dems, method="first")
+
+        # After merging LiDAR datasets set remaining NaN to no data/LiDAR
         # data_source: set areas with no values to No Data
-        dem.data_source.data[
-            numpy.isnan(dem.data_source.data)
-        ] = self.SOURCE_CLASSIFICATION["no data"]
+        dem.data_source.data = xarray.where(
+            dem.data_source.notnull(),
+            dem.data_source,
+            self.SOURCE_CLASSIFICATION["no data"],
+            keep_attrs=True,
+        )
+
         # lidar_source: Set areas with no LiDAR to "No LiDAR"
-        dem.lidar_source.data[
-            numpy.logical_not(
-                dem.data_source.data == self.SOURCE_CLASSIFICATION["LiDAR"]
-            )
-        ] = dataset_mapping["no LiDAR"]
-        # set any offshore values to ocean assuming drop offshore is selected
-        if (
-            self.catchment_geometry.foreshore_and_offshore.area.sum() > 0
-            and self.drop_offshore_lidar
-        ):
-            ocean_mask = numpy.logical_not(
-                numpy.isnan(
-                    dem.z.rio.clip(
-                        self.catchment_geometry.foreshore_and_offshore.geometry,
-                        drop=False,
-                    ).data
-                )
-            )
-            dem.data_source.data[ocean_mask] = self.SOURCE_CLASSIFICATION[
-                "ocean bathymetry"
-            ]
-            dem.lidar_source.data[ocean_mask] = dataset_mapping["no LiDAR"]
+        dem.lidar_source.data = xarray.where(
+            dem.lidar_source.notnull(),
+            dem.lidar_source,
+            dataset_mapping["no LiDAR"],
+            keep_attrs=True,
+        )
```

Comment:
hum... shouldn't it be dem["data_source"] = xarray.where(... ? (same as above)

Reply:
good catch thanks. Fixed